arch_x86.cpp

```diff
@@ -1927,7 +1927,42 @@ BNEndianness X86CommonArchitecture::GetEndianness() const
 
 vector<uint32_t> X86CommonArchitecture::GetGlobalRegisters()
 {
-	return vector< uint32_t> { XED_REG_CS, XED_REG_DS, XED_REG_ES, XED_REG_SS, XED_REG_FS, XED_REG_GS, XED_REG_FSBASE, XED_REG_GSBASE };
+	return vector< uint32_t> {
+		XED_REG_CS,
+		XED_REG_DS,
+		XED_REG_ES,
+		XED_REG_SS,
+		XED_REG_FS,
+		XED_REG_GS,
+		XED_REG_FSBASE,
+		XED_REG_GSBASE,
+		XED_REG_FLAGS,
+		XED_REG_EFLAGS,
+		XED_REG_RFLAGS
+	};
+}
+
+vector<uint32_t> X86CommonArchitecture::GetSystemRegisters()
```

Comment:
This change is good in general. However, there are other registers that can be considered system registers, e.g., DR0-DR7. I am not super familiar with the effect of GetSystemRegisters(), which I will discuss with the team and get back to you. It is likely some changes are needed here eventually

Reply:
I agree this list is not complete, it's just a step in the right direction.

As I understand it, system registers should be for anything where a read won't necessarily match what you write (if you even can write). Like control registers, where you write a command and read back a status. This designation should be used in binja to exclude these registers from both dataflow and parameter/return detection.

Reply:
You may have a look at https://github.com/Vector35/binaryninja-api/commit/ce385dc73e2d264fc2404c75b060a8a42486e27b, which explains the purpose of system registers. Also, this PR should best accompanied by some unit tests that verify the effectiveness of marking registers are system registers. I am not sure whether the general public can access our unit tests, so I can take over from here, i.e., add one or more commits to this PR and merge it later.

Reply:
And that you do not have to do more for this PR